sonar-db/src/test/java/org/sonar/db/CollationCheckerTest.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+import org.sonar.api.utils.MessageException;
+import org.sonar.db.dialect.MsSql;
+import org.sonar.db.dialect.MySql;
+import org.sonar.db.dialect.Oracle;
+import org.sonar.db.dialect.PostgreSql;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CollationCheckerTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  Database db = mock(Database.class, Mockito.RETURNS_MOCKS);
+  CollationChecker.StatementExecutor statementExecutor = mock(CollationChecker.StatementExecutor.class);
+  CollationChecker underTest = new CollationChecker(db, statementExecutor);
+
+  @Test
+  public void valid_oracle() throws Exception {
+    when(db.getDialect()).thenReturn(new Oracle());
+    answerSql(
+      singletonList(new String[] {"UTF8"}), singletonList(new String[] {"BINARY"}));
+
+    underTest.start();
+  }
+
+  @Test
+  public void fail_if_oracle_is_not_utf8() throws Exception {
+    expectedException.expect(MessageException.class);
+    expectedException.expectMessage("Oracle must be have UTF8 charset and BINARY sort. NLS_CHARACTERSET is LATIN and NLS_SORT is BINARY.");
+
+    when(db.getDialect()).thenReturn(new Oracle());
+    answerSql(
+      singletonList(new String[] {"LATIN"}), singletonList(new String[] {"BINARY"}));
+
+    underTest.start();
+  }
+
+  @Test
+  public void fail_if_oracle_is_not_case_sensitive() throws Exception {
+    expectedException.expect(MessageException.class);
```

Comment:
`expectedException` should be configured right before the statement supposed to raise the exception
better for clarity, avoid catching exception not raised at the expected time